WeakAuras/Features.lua

```diff
@@ -84,7 +100,7 @@ function Features:Register(feature)
     feature.sub = Private.CreateSubscribableObject()
     for _, buildType in ipairs(feature.autoEnable or {}) do
       if WeakAuras.buildType == buildType then
-        self:Enable(feature.id)
+        self:Enable(feature.id, true)
```

Comment:
cherry picked this from the undo pull request; otherwise it's annoying to disable a feature which auto-enables on your build type

Reply:
Hmm, this leads to the feature being enabled at first, before being disabled by hydrate.

I think a simpler approach is to require:
a) Registering a feature must be done before Hydrate
b) Hydrate decides on whether a feature is enabled/disabled, by whatever logic you decide.
.
That centralizes the logic on whether a feature starts off as enabled/disabled in one place.

Reply:
I'm not sure if it actually is simpler, but I see the appeal.

Reply:
I reworked Hydrate() to be a barrier between registering features, and setting their enable state. It's an error to try to register after hydration, or to enable before.